metadata/src/main/java/org/apache/kafka/controller/AclControlManager.java

```diff
@@ -64,12 +66,44 @@
  * completed, which is another reason the prepare / complete callbacks are needed.
  */
 public class AclControlManager {
+    static class Builder {
+        private LogContext logContext = null;
+        private SnapshotRegistry snapshotRegistry = null;
+        private Optional<ClusterMetadataAuthorizer> authorizer = Optional.empty();
+
+        Builder setLogContext(LogContext logContext) {
+            this.logContext = logContext;
+            return this;
+        }
+
+        Builder setSnapshotRegistry(SnapshotRegistry snapshotRegistry) {
+            this.snapshotRegistry = snapshotRegistry;
+            return this;
+        }
+
+        Builder setClusterMetadataAuthorizer(Optional<ClusterMetadataAuthorizer> authorizer) {
+            this.authorizer = authorizer;
+            return this;
+        }
+
+        AclControlManager build() {
+            if (logContext == null) logContext = new LogContext();
```

Comment:
Do you think it would be useful to add a LogPrefix here (and other places changed in this PR) such as 
```
LogContext("[AclControlManager id=nodeId")
```

Reply:
I agree this is a worthwhile question to consider.  I do note that the builders that already exist don't set anything special.  For example, from `ConfigurationControlManager.Builder.build()`:

```
            if (logContext == null) logContext = new LogContext();
```

And all of these builder log contexts do get explicitly set in QuorumController (including the new ones via this PR), so the default will never be used.

So my guess is the answer is no, it isn't worth doing.  Colin, WDYT?

Reply:
@rondagostino is correct that the defaults here do not get used except in unit tests.

In general we want `[QuorumController id=1] ` to appear before every quorum controller log message. The class name (AclControlManager or whatever) actually gets logged separately at the end of the line, when using the typical log4j setup.

Reply:
ok, sounds good.